chain/client/src/view_client.rs

```diff
@@ -172,12 +204,130 @@ impl ViewClientActor {
         finality: &Finality,
     ) -> Result<CryptoHash, near_chain::Error> {
         match finality {
-            Finality::None => Ok(self.chain.head()?.last_block_hash),
-            Finality::DoomSlug => Ok(*self.chain.head_header()?.last_ds_final_block()),
-            Finality::Final => Ok(self.chain.final_head()?.last_block_hash),
+            Finality::None => Ok(self.hot_chain.head()?.last_block_hash),
+            Finality::DoomSlug => Ok(*self.hot_chain.head_header()?.last_ds_final_block()),
+            Finality::Final => Ok(self.hot_chain.final_head()?.last_block_hash),
         }
     }
 
+    /// Determines temperature of a block with given block hash.
+    ///
+    /// Returned temperature allows fetching all data related to block at given
+    /// height as well as data about previous and next blocks (if any).
+    fn get_temperature(&self, _block_hash: &CryptoHash) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of a block with given block height.
+    ///
+    /// Returned temperature allows fetching all data related to block with
+    /// given hash as well as data about previous and next blocks (if any).
+    fn get_temperature_by_height(
+        &self,
+        _block_height: BlockHeight,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of a chunk with given chunk hash.
+    fn get_temperature_by_chunk_hash(
+        &self,
+        _chunk_hash: &ChunkHash,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of data from transaction hash.
+    fn get_temperature_by_tx_hash(
+        &self,
+        _tx_hash: &CryptoHash,
+        _sender_id: &AccountId,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of data from receipt hash.
+    fn get_temperature_by_rx_hash(
+        &self,
+        _rx_hash: &CryptoHash,
+        _receiver_id: Option<&AccountId>,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of a chunk with given chunk hash.
```

Comment:
Not sure; at the moment I don’t think we do though I imagine we might want to gc EpochInfo.  It may also be that this will need to return Hot or Cold depending whether all blocks in the epochs are hot or cold.